src/OpenTelemetry.ResourceDetectors.Host/HostDetector.cs

```diff
@@ -25,9 +25,9 @@ public Resource Detect()
                 new(HostSemanticConventions.AttributeHostName, Environment.MachineName),
             });
         }
-        catch (InvalidOperationException)
+        catch (InvalidOperationException ex)
```

Comment:
Can you elaborate on "Resource creation throwing" ?

Reply:
[Resource.cs](https://github.com/open-telemetry/opentelemetry-dotnet/blob/52a03b11aef3b75fdeaa4dd0fe8e9b1d71ea4e0a/src/OpenTelemetry/Resources/Resource.cs#L96) may throw exceptions, most of which are related to key validation. In this case, since the key is a constant, it shouldn't pose a problem. However, there is a scenario where the value is evaluated, and it's unclear how a null value would be handled in this context.

Reply:
In this case, the contract on the library is pretty solid - return not null value.

If you think that it is an issue, and we should not trust any external contracts on this, we should probably rewrite also all other resource detectors and catch AttributeNullException. IMO outside of the scope of this PR.